cmd/agentbasedinstaller/register.go

```diff
@@ -85,11 +108,54 @@ func RegisterInfraEnv(ctx context.Context, log *log.Logger, bmInventory *client.
 	if registerInfraEnvErr != nil {
 		return nil, registerInfraEnvErr
 	}
+
+	infraEnvID := infraEnvResult.Payload.ID.String()
+	log.Info("Registered infraenv with id: " + infraEnvID)
+
 	return infraEnvResult.Payload, nil
 }
 
+func RegisterExtraManifests(fsys fs.FS, ctx context.Context, log *log.Logger, client *manifests.Client, cluster *models.Cluster) error {
+
+	extras, err := fs.Glob(fsys, "*.y*ml")
+	if err != nil {
+		return err
+	}
+
+	if len(extras) == 0 {
+		return nil
+	}
+
+	log.Info("Registering extra manifests")
+
+	extraManifestsFolder := "openshift"
+
+	for _, extraManifestFileName := range extras {
+		bytes, err := fs.ReadFile(fsys, extraManifestFileName)
+		if err != nil {
+			return err
+		}
+
+		extraManifestContent := base64.StdEncoding.EncodeToString(bytes)
+		params := manifests.NewV2CreateClusterManifestParams().
+			WithClusterID(*cluster.ID).
+			WithCreateManifestParams(&models.CreateManifestParams{
+				FileName: &extraManifestFileName,
+				Folder:   &extraManifestsFolder,
+				Content:  &extraManifestContent,
+			})
+
+		_, err = client.V2CreateClusterManifest(ctx, params)
+		if err != nil {
+			return err
```

Comment:
Need to return `errorutil.GetAssistedError(err)` (see #4057)

Reply:
Done